Library/Homebrew/test/utils/git_spec.rb

```diff
@@ -0,0 +1,209 @@
+      allow_any_instance_of(Process::Status).to receive(:success?).and_return(true)
+      expect(described_class.git_available?).to be_truthy
+    end
+
+    it "returns false if git --version command does not succeed" do
+      allow_any_instance_of(Process::Status).to receive(:success?).and_return(false)
+      expect(described_class.git_available?).to be_falsey
+    end
+
+    it "returns git version if already set" do
+      described_class.instance_variable_set(:@git, true)
+      expect(described_class.git_available?).to be_truthy
+      described_class.instance_variable_set(:@git, nil)
+    end
+  end
+
+  describe "::git_path" do
+    context "when git is not available" do
+      before do
+        described_class.instance_variable_set(:@git, false)
+      end
+
+      it "returns" do
+        expect(described_class.git_path).to eq(nil)
+      end
+    end
+
+    context "when git is available" do
+      before do
+        described_class.instance_variable_set(:@git, true)
+      end
+
+      it "returns path of git" do
+        allow(Utils).to receive(:popen_read).with(HOMEBREW_SHIMS_PATH/"scm/git",
+                                                  "--homebrew=print-path")
+                                            .and_return("git")
+        expect(described_class.git_path).to eq("git")
+      end
+
+      it "returns git_path if already set" do
+        described_class.instance_variable_set(:@git_path, "git")
+        expect(described_class.git_path).to eq("git")
+        described_class.instance_variable_set(:@git_path, nil)
+      end
+    end
+  end
+
+  describe "::git_version" do
+    context "when git is not available" do
+      before do
+        described_class.instance_variable_set(:@git, false)
+      end
+
+      it "returns nil" do
+        expect(described_class.git_path).to eq(nil)
+      end
+    end
+
+    context "when git is available" do
+      before do
+        described_class.instance_variable_set(:@git, true)
+      end
+
+      it "returns version of git" do
+        expect(described_class.git_version).not_to be_nil
+      end
+
+      it "returns git_version if already set" do
+        described_class.instance_variable_set(:@git_version, "v1")
+        expect(described_class.git_version).to eq("v1")
+        described_class.instance_variable_set(:@git_version, nil)
+      end
+    end
+  end
+
+  describe "::git_remote_exists" do
+    context "when git is not available" do
+      before do
+        described_class.instance_variable_set(:@git, false)
+      end
+
+      it "returns true" do
+        expect(described_class.git_remote_exists("blah")).to be_truthy
+      end
+    end
+
+    context "when git is available" do
+      before(:all) do
+        @initial_var = described_class.instance_variable_get(:@git)
+        described_class.instance_variable_set(:@git, true)
+
+        git = HOMEBREW_SHIMS_PATH/"scm/git"
+        @repo = Pathname.new("hey")
```

Comment:
Realised I can do away with the before(:all) block. Hence, removed it. Thanks :)